### PR TITLE
Timestamp in EventEnvelope, #28331

### PR DESCRIPTION
**Pull request by raboof**
* added to EventEnvolope and therefore include case class stuff
  for binary compatibility
* also added in PersistentRepr, which for example is the serialized format in
  LeveldbJournal

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #28331

**Comment by patriknw** [comment]
Backport of https://github.com/akka/akka/pull/28383

File: akka-persistence/src/main/java/akka/persistence/serialization/MessageFormats.java [hunk 1]
```diff
@@ -140,6 +140,16 @@ public static void registerAllExtensions(
      */
```
Comment (patriknw) [inline akka-persistence/src/main/java/akka/persistence/serialization/MessageFormats.java]: did you re-generate this, with the right protoc version? we have different in master in now

**Review by patriknw — COMMENTED** [review]
thanks for taking care of the backport, one question...

File: akka-persistence/src/main/java/akka/persistence/serialization/MessageFormats.java (hunk 1 above)
Comment (raboof) [reply akka-persistence/src/main/java/akka/persistence/serialization/MessageFormats.java]: I regenerated with 2.5, yes

**Review by patriknw — APPROVED** [review]

**Review by johanandren — APPROVED** [review]

**Pull request merged**